nmdc_automation/re_iding/scripts/re_id_tool.py

```diff
@@ -808,5 +913,114 @@ def _get_has_input_from_read_qc(api_client, legacy_id):
+    logging.info(f"Updating {legacy_study_id} /  {study_record['id']}: {study_record['name']}")
+
+    # Copy the legacy ID to gold_study_identifiers if it is not already there
+    if legacy_study_id.startswith("gold:"):
+        gold_ids = study_record.get("gold_sequencing_project_identifiers", [])
+        if legacy_study_id not in gold_ids:
+            gold_ids.append(legacy_study_id)
+            study_record["gold_sequencing_project_identifiers"] = gold_ids
+            logging.info(f"Added legacy study ID to gold_sequencing_project_identifiers: {legacy_study_id}")
+
+    if no_update:
+        logging.info(f"Skipping database update")
+    else:
+        logging.info(f"Updating database with new study record")
+        result = db_client["study_set"].replace_one({"id": legacy_study_id}, study_record)
+        logging.info(f"Updated {result.modified_count} study_set records")
+    return study_record
+
+def _update_biosample_record(biosample_record: dict, new_study_id: str, db_client: Database[Union[Mapping[str, Any], Any]],
+                             api_client: NmdcRuntimeApi,  no_update: bool) -> dict:
+    """
+    Update the biosample record with the new ID
+    """
+    legacy_biosample_id = biosample_record["id"]
+    # Update the part_of array with the new study ID
+    part_of = biosample_record.get("part_of", [])
+    # Remove gold: IDs from part_of array
+    part_of = [id for id in part_of if not id.startswith("gold:")]
+    if new_study_id not in part_of:
+        part_of.append(new_study_id)
+        biosample_record["part_of"] = part_of
+        logging.info(f"Added new study ID to part_of: {new_study_id}")
+
+    # Add the legacy ID to gold_biosample_identifiers if it is not already there
+    if legacy_biosample_id.startswith("gold:"):
+        gold_ids = biosample_record.get("gold_biosample_identifiers", [])
+        if legacy_biosample_id not in gold_ids:
+            gold_ids.append(legacy_biosample_id)
+            biosample_record["gold_biosample_identifiers"] = gold_ids
+            logging.info(f"Added legacy biosample ID to gold_biosample_identifiers: {legacy_biosample_id}")
+
+    # Mint a new biosample ID if needed
+    if not biosample_record["id"].startswith("nmdc:bsm-"):
+        new_biosample_id = api_client.minter("nmdc:Biosample")
+        biosample_record["id"] = new_biosample_id
+        logging.info(f"Minted new biosample ID: {new_biosample_id}")
+    if no_update:
+        logging.info(f"Skip Update:  {legacy_biosample_id} /  {biosample_record['id']} : {biosample_record['name']}")
+    else:
+        result = db_client["biosample_set"].replace_one({"id": legacy_biosample_id}, biosample_record)
+        logging.info(f"Updated {result.modified_count} biosample_set records {legacy_biosample_id} /  {biosample_record['id']} : {biosample_record['name']}")
+    return biosample_record
+
+def _update_omics_processing_record(omics_processing_record: dict,new_study_id: str, new_biosample_id: str, db_client: (
```

Comment:
This function is missing logic handle emsl: prefixes. Please see original code I wrote
https://github.com/microbiomedata/nmdc-schema/blob/55f087ad90982d647b82f42efd2662f8368740ba/nmdc_schema/napa_study_biosample_omics_migration.py#L167

Reply:
looks good now.